### ISSUE TITLE
Backups accumulate after TTL expire

### ISSUE THREAD

**Issue by fmacicasan** [hazelcast/hazelcast#8939]
Hello!

We are using the map put API with TTL and see constantly growing backup entry counts and backup heap usage.

We were expecting that the evict would propagate to the backup. Isn't this the case?

We use a 2 node cluster with hazelcast version 3.3.3. I added a test which underlines this scenario. Tested on both 3.3.3 and 3.7.1.

https://gist.github.com/fmacicasan/81d873e0fc60f57539b4ed17e542b9a3

Also adding the management console snippet that shows this issue. 
![image](https://cloud.githubusercontent.com/assets/902751/18719469/25efd52c-8030-11e6-873c-74d669f31441.png)

Thanks,
Florin

**Comment by ahmetmircik** [comment]
Hi @fmacicasan, expiration happens gradually by a background task, test should be written to make an eventual check of number of entries. Slightly modified your test and it seems backups also swept eventually. Also you can see my answer [here](https://github.com/hazelcast/hazelcast/issues/8894#issuecomment-247859168) for speeding up that background sweeper task.

```
public class Test {

    public static void main(String[] args) {
        Config config = new Config();
        HazelcastInstance instance = Hazelcast.newHazelcastInstance(config);
        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);

        IMap map1 = instance.getMap("m");
        IMap map2 = instance2.getMap("m");

        for (int i = 0; i < 10; i++) {
            map1.put("key" + i, "value" + i, 5, TimeUnit.SECONDS);
        }

        long start = System.nanoTime();
        long wait = TimeUnit.SECONDS.toNanos(30);

        while (true) {
            long totalEntryCount = getTotalEntryCount(map1) + getTotalEntryCount(map2);

            if (totalEntryCount != 0 && System.nanoTime() - start < wait ) {
                LockSupport.parkNanos(TimeUnit.SECONDS.toNanos(1));
            } else {
                System.out.println("totalCount = " + totalEntryCount);
                System.exit(0);
            }
        }
    }

    protected static long getTotalEntryCount(IMap m) {
        LocalMapStats localMapStats = m.getLocalMapStats();
        return localMapStats.getOwnedEntryCount() + localMapStats.getBackupEntryCount();
    }
}

```

**Comment by fmacicasan** [comment]
When was this backup sweep introduced? I ran your test on 3.3.3 and there are still 10 entries after the 30 seconds.

**Comment by ahmetmircik** [comment]
Introduced by starting 3.3, but the issue with backups ([4786](https://github.com/hazelcast/hazelcast/issues/4786)) was fixed in 3.4.1. Fix is not available in any 3.3.x patches. A higher version should be used. I recommend you 3.7.1.

**Comment by fmacicasan** [comment]
Thanks @ahmetmircik. I confirm that the backups get swept in the version we are looking to upgrade to.

Thanks for your assistance!
